source/ast.rs

```diff
@@ -43,15 +43,15 @@ pub struct Term {
 pub struct Binding {
     ///
     pub variable: String,
-    /// 
+    ///
     pub expression: Term
 }
 
-/// An addition of two terms.
 #[derive(Debug, PartialEq)]
-pub struct Addition {
-    /// The left-hand side of the addition.
-    pub a: Term,
-    /// The right-hand side of the addition.
-    pub b: Term
+pub enum Expression {
+    Multiplication { a: Term, b: Term },
```

Comment:
Actually, the multiplication (or other arithmetic operator) members are of kind `Expression`. This is the tricky pat. This is not possible to compute the size of the enum if we do:
```rust
pub enum Expression {
    Multiplication { a: Expression, b: Expression },
    …
}
```
We will get the following error:
```
error: recursive type `Expression` has infinite size [--explain E0072]
 --> <anon>:3:1
3 |> pub enum Expression {
  |> ^
```
(see https://doc.rust-lang.org/error-index.html#E0072)
So the solution is to use a [`Box`](https://doc.rust-lang.org/std/boxed/struct.Box.html), and so:
```rust
pub enum Expression {
    Multiplication { a: Box<Expression>, b: Box<Expression> },
    …
}
```
and it will work. See the documentation for `Box` to understand why it removes the “infinite” size of the enum.